apps/nova/src/components/UpcomingLessons/Content.tsx

```diff
@@ -26,7 +26,8 @@ export const Content: React.FC<{
   error: boolean;
   hasMore: boolean;
   more: Void;
-}> = ({ list, loading, fetching, error, hasMore, more }) => {
+  refetch: Void;
```

Comment:
The type `Void` bestows ambiguity into the code. It'll be tedious for newcomer teammates. At least we can add a comment describing the expected values to be passed.

Reply:
The type `Void` is exactly the same as `() => void`. It doesn't accept anything. **The main idea behind it is represent a simple callback handler that doesn't accept any params**. 

I think the problem is with the name. It conflicts with the native type `void` (lower case). Maybe we can rename the type into something else like `Callback` or not use it at all and just type `() => void`. 

Or do it this way (the best in my opinion). 

```suggestion
  refetch(): void;
```

cc @mmoehabb @mostafakamar2308

Reply:
You didn't get what I mean. Both `Void` and `() => void` bestow ambiguity. When I first read this code I didn't easily figure out what value should I pass the parameter `retry`. We may define more specific type or add a simple comment, for instance: `// A callback function that invokes tanstack query refetch method`.

Reply:
Maybe the best solution is to define a "dump" type in the types dir:
```TS
/**
 * A callback function that invokes tanstack query refetch method
 */
type TanStackRefetchCallback = Void;
```
what do you think? @neuodev

Reply:
> The child component should not be considered with the parent implementation 

In this case, the `Content` component should not be aware of how the parent is fetching the data. For example, the parent should use what ever approach possible (e.g., tanstack query, Fetch API, axios, ...etc).